server/plugins/schema-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/schema/obmysql/parser/OBMySQLGetDBTableByParser.java

```diff
@@ -244,33 +246,102 @@ public DBTablePartition getPartition() {
         if (partitionStmt.getSubPartitionOption() == null) {
             return partition;
         }
-        // TODO 目前 ODC 仅支持 HASH/KEY 二级分区, 其它类型后续需补充
         partition.setSubpartitionTemplated(partitionStmt.getSubPartitionOption().getTemplates() != null);
-        SubPartitionOption subOption = partitionStmt.getSubPartitionOption();
         String type = partitionStmt.getSubPartitionOption().getType();
         if ("key".equals(type.toLowerCase())) {
-            subPartitionOption.setType(DBTablePartitionType.KEY);
-            subPartitionOption.setColumnNames(subOption.getSubPartitionTargets() == null ? null
-                    : subOption.getSubPartitionTargets().stream().map(item -> removeIdentifiers(item.getText()))
-                            .collect(Collectors.toList()));
-            subPartitionOption
-                    .setPartitionsNum(partition.getSubpartitionTemplated() ? subOption.getTemplates().size()
-                            : partitionStmt.getPartitionElements().get(0).getSubPartitionElements().size());
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt, DBTablePartitionType.KEY,
+                    false);
         } else if ("hash".equals(type.toLowerCase())) {
-            subPartitionOption.setType(DBTablePartitionType.HASH);
-            Expression expression = subOption.getSubPartitionTargets().get(0);
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt, DBTablePartitionType.HASH,
+                    true);
+        } else if ("range".equals(type.toLowerCase())) {
```

Comment:
Could we eliminate these `if` branches? Actually, your `parseSubPartition` method only cares about the `DBTablePartitionType` and if it supports `expression`. 
1. May be you could convert the String to Enum in the Enum itself, like do it in  `DBTablePartitionType#fromValue`.
2. The answer to whether the partition type supports expression could also be implied in the `DBTablePartitionType`.
In this way, you could eliminate all the if braches and directly call `parseSubPartition`.

Reply:
ok